framework/src/timesteppers/TimeSequenceStepper.C

```diff
@@ -0,0 +1,83 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "TimeSequenceStepper.h"
+#include "FEProblem.h"
+#include "Transient.h"
+
+template<>
+InputParameters validParams<TimeSequenceStepper>()
+{
+  InputParameters params = validParams<TimeStepper>();
+  params.addRequiredParam<std::vector<Real> >("time_knots", "The values of t");
+  return params;
+}
+
+TimeSequenceStepper::TimeSequenceStepper(const InputParameters & parameters) :
+    TimeStepper(parameters),
+    _current_step(declareRestartableData("current_step", (unsigned int) 1)),
+    _time_knots(getParam<std::vector<Real> >("time_knots"))
+{
+  // there must be at least two time knots present
+  if (_time_knots.size() < 2)
+    mooseError("You must suppy at least two time knots.");
+
+  // sync _executioner.startTime with _time_knots
+  Real start_time = _executioner.getStartTime();
+  if (MooseUtils::relativeFuzzyGreaterEqual(start_time, _time_knots.back()))
+    mooseError("Start time is later than latest time_knot.");
+
+  if (MooseUtils::relativeFuzzyLessThan(start_time, _time_knots[0]))
+    _time_knots.insert(_time_knots.begin(), start_time);
+  else
+  {
+    std::vector<Real> temporary_time_knots(1, start_time);
+    for (unsigned int j = 0; j < _time_knots.size(); j++)
+      if (MooseUtils::relativeFuzzyLessThan(start_time, _time_knots[j]))
+        temporary_time_knots.push_back(_time_knots[j]);
+    _time_knots = temporary_time_knots;
+  }
+
+  if (_app.halfTransient())
+  {
+    unsigned int half = (_time_knots.size() - 1) / 2;
+    _executioner.endTime() = _time_knots[half];
+  }
+  else
+    _executioner.endTime() = _time_knots.back();
+}
+
+void
+TimeSequenceStepper::init()
+{
+}
+
+void
+TimeSequenceStepper::step()
+{
+  TimeStepper::step();
+  _current_step++;
```

Comment:
How does this handle a failed solve? I don't see any way the current step can get backed up or retried.